### DISCUSSION TITLE
support vanilla JS in client directives

### DISCUSSION THREAD

**Discussion by endymion1818** [withastro/roadmap#774] [Proposal]
### Body

# Summary

In addition to supporting client:* directives for popular frameworks such as React and Lit, I would like to see support for vanilla JavaScript.

# Background & Motivation

I have a library (a media player), which I have added to the project's dependencies in package.json. I would like to be able to access that package on the client only.

# Goals

- To be able to use NPM modules and others that have already been compiled by the browser and installed as dependencies 
- To be able to use NPM modules without a framework

# Example

```
---
import Something from '@namespace/something'
---
<Something client:only="vanilla" />
```

# Similar issues

https://github.com/withastro/roadmap/discussions/102

**Comment by MoustaphaDev** [comment]
I'm not sure I understand your proposal, and it seems to me that it diverges quite a lot from #102.
You're saying that you would like to be able to access that package on the client only. It's already possible to do that! You can use [`script` tags](https://docs.astro.build/en/guides/client-side-scripts/#using-script-in-astro) for that.
Let me know if that's what you were looking for 🙂

**Reply by endymion1818** [reply]
Hey thanks for the reply, yes I think I did achieve this ultimately with that method, sorry for not closing this one.

As a side note I think it's really interesting that this could be possible natively soon with import maps: https://www.honeybadger.io/blog/import-maps/
